### PR TITLE
frontend: Migrate CertificateAuthority from WithClusterConfig to Connect and use Radio component

### PR DESCRIPTION
**Pull request by kyoto**
This change causes the following to be added to the `.progress` file when the CA Certificate and CA Private Key fields are not set.

```
"clusterConfig": {
  ...
  "ignore": {
     ...
     "caCertificate": true,
     "caPrivateKey": true
```

### CODE REVIEW

File: installer/frontend/cluster-config.js [hunk 1]
```diff
@@ -176,7 +177,7 @@ export const DEFAULT_CLUSTER_CONFIG = {
   [BM_TECTONIC_DOMAIN]: '',
   [CA_CERTIFICATE]: '',
   [CA_PRIVATE_KEY]: '',
-  [CA_TYPE]: 'self-signed',
+  [CA_TYPE]: '',
```
Comment (kans) [inline installer/frontend/cluster-config.js]: I don't think you need this at all anymore.

File: installer/frontend/components/certificate-authority.jsx [hunk 2]
```diff
@@ -1,102 +1,70 @@
...
+import { CA_CERTIFICATE, CA_PRIVATE_KEY, CA_TYPE, CA_TYPES } from '../cluster-config';
+import { Field, Form } from '../form';
 import { validate } from '../validate';
-import { CA_TYPE, CA_CERTIFICATE, CA_PRIVATE_KEY } from '../cluster-config';
-import { WithClusterConfig, CertArea, PrivateKeyArea } from './ui';
+import { CertArea, Connect, PrivateKeyArea, Radio } from './ui';
+
+const form = new Form('CERTIFICATE_AUTHORITY', [
+  new Field(CA_TYPE, {default: CA_TYPES.SELF_SIGNED}),
+  new Field(CA_CERTIFICATE, {default: '', validator: validate.certificate}),
```
Comment (kans) [inline installer/frontend/components/certificate-authority.jsx]: There is a way to tell `CA_CERTIFICATE` to look at `CA_TYPE` and optionally enable it.

File: installer/frontend/cluster-config.js (hunk 1 above)
Comment (kyoto) [reply installer/frontend/cluster-config.js]: Apparently it's still needed for the unit tests.

File: installer/frontend/components/certificate-authority.jsx (hunk 2 above)
Comment (kyoto) [reply installer/frontend/components/certificate-authority.jsx]: Thanks, I added `ignoreWhen` callbacks for the cert and key fields and updated the PR description.

File: installer/frontend/cluster-config.js (hunk 1 above)
Comment (kans) [reply installer/frontend/cluster-config.js]: Oh, I think you just need to import the file in the test suite.

File: installer/frontend/components/certificate-authority.jsx [hunk 3]
```diff
@@ -1,102 +1,78 @@
...
-  if (clusterConfig[CA_TYPE] === 'self-signed') {
-    return true;
-  }
+  </div>
+</div>);
 
-  return (!validate.certificate(clusterConfig[CA_CERTIFICATE]) &&
-          !validate.privateKey(clusterConfig[CA_PRIVATE_KEY]));
-};
+CertificateAuthority.canNavigateForward = state => state.clusterConfig[CA_TYPE] === CA_TYPES.SELF_SIGNED || form.canNavigateForward(state);
```
Comment (kans) [inline installer/frontend/components/certificate-authority.jsx]: You should be able to remove this logic now.

File: installer/frontend/cluster-config.js (hunk 1 above)
Comment (kyoto) [reply installer/frontend/cluster-config.js]: Actually the tests do pass, but they start to output "Removed clusterConfig.caType because it's not in defaults." with `console.error`. Maybe the tests should be changed? I'm thinking that shouldn't be part of this PR though.

File: installer/frontend/components/certificate-authority.jsx (hunk 3 above)
Comment (kyoto) [reply installer/frontend/components/certificate-authority.jsx]: OK, I changed it so that this is now just `CertificateAuthority.canNavigateForward = form.canNavigateForward` and all the validation is done in the form level `validator`.

**Comment by kyoto** [comment]
@kans Addressed your review comments.

**Comment by kyoto** [comment]
ok to test

File: installer/frontend/components/etcd.jsx [hunk 4]
```diff
@@ -76,7 +76,7 @@ export const Etcd = connect(({clusterConfig}) => ({
               <Connect field={ETCD_OPTION}>
                 <Radio name={ETCD_OPTION} value={ETCD_OPTIONS.EXTERNAL} id={ETCD_OPTIONS.EXTERNAL} />
               </Connect>
-            I have my own v3 etcd cluster
+              I have my own v3 etcd cluster
```
Comment (squat) [inline installer/frontend/components/etcd.jsx]: Not part of this pr but I am just noticing that this should really be "etcd v3" rather than the other way around.

Comment (kyoto) [reply installer/frontend/components/etcd.jsx]: Fixed: #2019
